tools/health/health_checker_pool.go

```diff
@@ -247,3 +247,103 @@ func (ncp *NodeHealthCheckerPool) WaitAllHashesEqual(height uint64) error {
 		}
 	}
 }
+
+func (ncp *NodeHealthCheckerPool) FindInconsistentHashesAtHeight(height uint64) map[string]sdk.Hash {
```

Comment:
I think you can use the `WaitAllHashesEqual` because it already compares hashes and log if something is wrong